### PR TITLE
refactor: Remove project level method that are per environment

### PR DESCRIPTION
**Pull request by olivier-lacroix**

### CODE REVIEW

**Review by baszalmstra — APPROVED** [review]
Thanks for fixing technical debt!

File: src/cli/add.rs [hunk 1]
```diff
@@ -376,7 +371,11 @@ async fn determine_latest_versions(
 ) -> miette::Result<Vec<Version>> {
     // Get platforms to search for including NoArch
     let platforms = if platforms.is_empty() {
-        let mut temp = project.platforms().into_iter().collect_vec();
+        let mut temp = project
```
Comment (olivier-lacroix) [inline src/cli/add.rs:374]: shouldn't latest versions be determined by environment?

File: src/cli/add.rs [hunk 2]
```diff
@@ -125,14 +125,9 @@ pub async fn execute(args: Args) -> miette::Result<()> {
     verify_prefix_location_unchanged(project.default_environment().dir().as_path()).await?;
 
     // Add the platform if it is not already present
-    let platforms_to_add = spec_platforms
```
Comment (olivier-lacroix) [inline src/cli/add.rs:128]: platforms are properly deduplicated in `manifest.add_platforms` already

File: src/cli/add.rs (hunk 1 above)
Comment (ruben-arts) [reply src/cli/add.rs:374]: Yes they should, the determine function is heavily outdated. @baszalmstra is going to work on it next week.

**Pull request merged**